packages/snaps-execution-environments/src/common/endowments/network.ts

```diff
@@ -144,11 +211,14 @@ const createNetwork = () => {
       res = new ResponseWrapper(
         await withTeardown(fetchPromise, teardownRef),
         teardownRef,
+        onStart,
+        onFinish,
       );
     } finally {
       if (openFetchConnection !== undefined) {
         openConnections.delete(openFetchConnection);
       }
+      notify({ method: 'OutboundResponse', params: { source: 'fetch' } });
```

Comment:
I'm thinking what would happen if `fetch` gets stuck or takes way too long to finish. It seems like the timer could be paused forever in that case? Do you maybe have a plan to handle something like that on the controller side? 

I would suggest that we have at least some safety mechanism if this edge case is possible. Maybe identify the request with params on the controller side and still set timeout for it, but one with the larger time frame.

Also, I'm thinking if this process can or should be surfaced to the users who will be waiting for `fetch` to complete. Might be complicated, but having it hooked to the UI somehow and displaying non-deterministic progress bar at least would contribute to better UX.

Reply:
Resolved in other conversations on Slack. Identified as non-major concern.